### ISSUE TITLE
Differentially private CIFAR 10 not training

### ISSUE THREAD

**Issue by samdow** [pytorch/functorch#467]
When I run [`examples/dp_cifar10/cifar10_transforms.py`](https://github.com/pytorch/functorch/blob/main/examples/dp_cifar10/cifar10_transforms.py) locally, I see that for each epoch the loss gets higher and higher and the accuracy stays below 10% (which it should be, approximately, if we just kept guessing 0 because there's 10 classes). Given that we have a passing test that checks per sample gradients and that I'm not seeing this happen with Opacus, this is suggesting that our recreation of the clipping + noising of the per sample gradients is incorrect. The task is to diagnose and fix this

Output:
```
  Train Epoch: 1  Loss: 7.328635 Acc@1: 0.000000 
  Train Epoch: 1  Loss: 37673.759659 Acc@1: 0.088727 
  Train Epoch: 1  Loss: 57173.875967 Acc@1: 0.087619 
  Train Epoch: 1  Loss: 94578.966633 Acc@1: 0.090323 
  Train Epoch: 1  Loss: 126669.624162 Acc@1: 0.085756 
  Train Epoch: 1  Loss: 139614.602267 Acc@1: 0.084706 
  Train Epoch: 1  Loss: 146442.156583 Acc@1: 0.082426 
  Train Epoch: 1  Loss: 154726.866043 Acc@1: 0.080845 
  Train Epoch: 1  Loss: 166711.730864 Acc@1: 0.080840 
  Train Epoch: 1  Loss: 176655.394025 Acc@1: 0.082593 
  Train Epoch: 1  Loss: 179217.489217 Acc@1: 0.083010 
  Train Epoch: 1  Loss: 186834.898972 Acc@1: 0.082739 
  Train Epoch: 1  Loss: 194612.639889 Acc@1: 0.082347 
  Train Epoch: 1  Loss: 207030.608934 Acc@1: 0.082626 
  Train Epoch: 1  Loss: 204050.683378 Acc@1: 0.083603 
  Train Epoch: 1  Loss: 195366.205116 Acc@1: 0.079735 
  Train Epoch: 1  Loss: 185335.132222 Acc@1: 0.074783 
  Train Epoch: 1  Loss: 175120.476002 Acc@1: 0.070409 
  Train Epoch: 1  Loss: 166249.561392 Acc@1: 0.066519 
  Train Epoch: 1  Loss: 157987.407736 Acc@1: 0.063058 
```

**Comment by spliew** [comment]
The calculated noisy gradient needs to be divided by `batch_size` here:

https://github.com/pytorch/functorch/blob/23e8c6641b4f938b98dc4f510c300c0a3a0555c4/examples/dp_cifar10/cifar10_transforms.py#L79

The analogous operation in Opacus is performed by the `scale_grad()` [method](https://github.com/pytorch/opacus/blob/c9abdcf1a6180965b427421eb019550da76a03ff/opacus/optimizers/optimizer.py#L434).
I am able to train with high accuracy by fixing the code with something like `param.grad = param_grad/batch_size`

**Comment by samdow** [comment]
Great catch @spliew! We'd welcome a PR if you want to make the change (especially if you can share the loss/accuracy you're seeing locally) or I can make it

**Comment by samdow** [comment]
Closing, thanks again @spliew!
